packages/babel-register/src/node.js

```diff
@@ -108,7 +108,10 @@ export default function register(opts?: Object = {}) {
   opts = Object.assign({}, opts);
   if (opts.extensions) hookExtensions(opts.extensions);
 
-  if (opts.cache === false) cache = null;
```

Comment:
Should we consider changing this to

    if (opts.cache === false && cache) {
      registerCache.clear();

Reply:
Done!